packages/frontend/src/features/sqlRunner/components/Sidebar.tsx

```diff
@@ -30,9 +30,13 @@ export const Sidebar: FC<Props> = ({ setSidebarOpen }) => {
         error,
     } = useRefreshTables({ projectUuid });
 
-    const { selectedChartType, activeSidebarTab, sqlColumns } = useAppSelector(
-        (state) => state.sqlRunner,
+    const selectedChartType = useAppSelector(
+        (state) => state.sqlRunner.selectedChartType,
     );
+    const activeSidebarTab = useAppSelector(
+        (state) => state.sqlRunner.activeSidebarTab,
+    );
+    const sqlColumns = useAppSelector((state) => state.sqlRunner.sqlColumns);
```

Comment:
small optimisation

Reply:
```suggestion
    const { selectedChartType, activeSidebarTab, sqlColumns } = useAppSelector(
        (state) => ({
            selectedChartType: state.sqlRunner.selectedChartType,
            activeSidebarTab: state.sqlRunner.activeSidebarTab,
            sqlColumns: state.sqlRunner.sqlColumns,
        }),
    );
```

By selecting only the necessary parts of the state, we can avoid potential unnecessary re-renders. The original was just missing the selection of child objects

Reply:
On second thought it now no longer fully couples the param name to the type which could cause issues later if we rename things. so feel free to ignore my suggestion if that is a concern

Reply:
> By selecting only the necessary parts of the state, we can avoid potential unnecessary re-renders. The original was just missing the selection of child objects

the suggested approach is returning always a new reference after an action dispatch, causing unnecessary rerenders after action dispatches. This was really useful: https://redux.js.org/usage/deriving-data-selectors#optimizing-selectors-with-memoization see the first: `completedTodos` example. 
```
The default comparison is a strict === reference comparison.
```
https://react-redux.js.org/api/hooks#equality-comparisons-and-updates

We can use another package like `reselect` to change the comparison logic tho 👍